plinth/modules/bind/__init__.py

```diff
@@ -0,0 +1,142 @@
+"""
+
+from django.urls import reverse_lazy
+from django.utils.translation import ugettext_lazy as _
+
+from plinth import actions
+from plinth import action_utils
+from plinth import cfg
+from plinth import frontpage
+from plinth import service as service_module
+from plinth.views import ServiceView
+
+
+version = 1
+
+depends = ['apps']
+
+title = _('Domain Name Server \n (BIND)')
+
+service = None
+
+managed_services = ['bind9']
+
+managed_packages = ['bind9', 'bind9utils', 'bind9-doc']
+
+description = [
+    _('BIND is open source software that enables you to publish your Domain '
+      'Name System (DNS) information on the Internet, and to resolve '
+      'DNS queries for your users.'),
+
+    _('BIND is by far the most widely used DNS software on the Internet, '
+      'providing a robust and stable platform on top of which organizations'
+      ' can build distributed computing systems with the knowledge that those '
+      'systems are fully compliant with published DNS standards.')
+]
+
+CONFIG_FILE = '/etc/bind/named.conf.options'
+
+value1 = 'acl goodclients { \n   localhost;\n};\n'
+value2 = '        recursion yes;\n           allow-query { goodclients; };\n\n'
+value3 = '	// 	8.8.8.8;\n	// 	8.8.4.4;\n'
+
+
+def init():
+    """Intialize the BIND module."""
+    menu = cfg.main_menu.get('system:index')
+    menu.add_urlname(title, 'glyphicon-globe', 'bind:index')
+
+    global service
+    setup_helper = globals()['setup_helper']
+    if setup_helper.get_state() != 'needs-setup':
+        service = service_module.Service(
+            managed_services[0], title, ports=['bind-plinth'],
+            is_external=True,
+            )
+
+
+class BindServiceView(ServiceView):
+    service_id = managed_services[0]
+    diagnostics_module_name = "bind"
+    description = description
+
+
+def setup(helper, old_version=None):
+    """Install and configure the module."""
+    helper.install(managed_packages)
+    global service
+    if service is None:
+        service = service_module.Service(
+            managed_services[0], title, ports=['bind-plinth'],
+            is_external=True,
+            enable=enable, disable=disable)
+    helper.call('post', service.notify_enabled, None, True)
+    helper.call('post', default_config)
+
+
+def enable():
+    """Enable the module."""
+    actions.superuser_run('service', ['enable', managed_services[0]])
+    add_shortcut()
+
+
+def disable():
+    """Disable the module."""
+    actions.superuser_run('service', ['disable', managed_services[0]])
+    frontpage.remove_shortcut('bind')
+
+
+def diagnose():
+    """Run diagnostics and return the results."""
+    results = []
+
+    results.append(action_utils.diagnose_port_listening(53, 'tcp6'))
+    results.append(action_utils.diagnose_port_listening(53, 'udp6'))
+
+    return results
+
+def default_config():
+    """Initialize config file for BIND"""
+    actions.superuser_run('bind', ['setup'])
+    f = open(CONFIG_FILE, "r")
```

Comment:
Right now, the ```bind setup``` line has no effect (if it runs at all?)--because setup isn't implemented in the bind action script.

I added some comments where we need to implement it.